src/utils/utils.ts

```diff
@@ -1,65 +1,93 @@
 export const cleanMarketsData = (
-  response: { id: string; marketname: any }[], distance: number
-) => {
+  response: { id: string; marketname: string }[], distance: number ) => {
   let mapped = response.map(currentMarket => {
     let name = currentMarket.marketname.split(' ');
-    let distance = name[0];
+    let distance = parseInt(name[0]);
     return {
       id: parseInt(currentMarket.id),
       distanceFromZip: Math.round(distance * 10) / 10,
-      marketName: name.slice(1).join(' ')
+      marketName: name.slice(1).join(' '),
+      schedule: {}
     };
   });
 
   return mapped.filter(currentMarket => currentMarket.distanceFromZip < distance);
 };
 
-export const cleanDetailsData = (
-  arrayOfPromises: {
-    marketdetails: {
-      GoogleLink: string;
-      Address: string;
-      Schedule: string;
-      Products: string;
-    };
-  }[]
-) => {
-  return arrayOfPromises.map(currentMarket => {
-    let addressArray = currentMarket.marketdetails.Address.split(', ');
+export const cleanDetailsData = (response: {
+    GoogleLink: string;
+    Address: string;
+    Schedule: string;
+    Products: string;
+  }, id: number) => {
 
-    let schedule = currentMarket.marketdetails.Schedule.replaceAll(
-      '<br>',
-      ''
-    ).split(';');
-    schedule.pop();
+  let addressArray = response.Address.split(', ');
+  let schedule = response.Schedule.replaceAll('<br>','').split(';');
+  schedule.pop();
 
-    let formattedSchedule = schedule.map(currentDay => {
-      let season;
-      if (currentDay.split(': ')[0].slice(0, -4).length < 24) {
-        season = 'bad format';
-      } else {
-        season = currentDay.split(': ')[0].slice(0, -4);
-      }
-
-      return {
-        dayOfWeek: currentDay.split(': ')[0].slice(-3),
-        time: currentDay.split(': ')[1],
-        season: season
-      };
-    });
+  let formattedSchedule = schedule.map(currentDay => {
+    let season;
+    if (currentDay.split(': ')[0].slice(0, -4).length < 24) {
+      season = 'bad format';
+    } else {
+      season = currentDay.split(': ')[0].slice(0, -4);
+    }
 
     return {
-      street: addressArray[0],
-      city: addressArray[addressArray.length - 3],
-      state: addressArray[addressArray.length - 2],
-      zip: addressArray[addressArray.length - 1],
-      schedule: formattedSchedule,
-      products: currentMarket.marketdetails.Products.split(';'),
-      mapsLink: currentMarket.marketdetails.GoogleLink
+      dayOfWeek: currentDay.split(': ')[0].slice(-3),
+      time: currentDay.split(': ')[1],
+      season: season
     };
   });
+
+  return {
+    id: id,
+    street: addressArray[0],
+    city: addressArray[addressArray.length - 3],
```

Comment:
Is there a better/cleaner way to access the indexes out of the addressArray than .length minus number?

Reply:
length + number, but both have the same Big 0 cost.